src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            fixed (byte* bytePointer = &b)
+            {
+                // Note: It's important that this switch handles lengths at least up to 22.
+                // See notes below near the main loop for why.
+
+                // The switch will be very fast since it can be implemented using a jump
+                // table in assembly. See http://stackoverflow.com/a/449297/4077294 for more info.
+
+                switch (byteLength)
+                {
+                    case 1:
+                        *bytePointer = (byte)0;
+                        return;
+                    case 2:
+                        *(short*)bytePointer = (short)0;
+                        return;
+                    case 3:
+                        *(short*)bytePointer = (short)0;
+                        *(bytePointer + 2) = (byte)0;
+                        return;
+                    case 4:
+                        *(int*)bytePointer = (int)0;
+                        return;
+                    case 5:
+                        *(int*)bytePointer = (int)0;
+                        *(bytePointer + 4) = (byte)0;
+                        return;
+                    case 6:
+                        *(int*)bytePointer = (int)0;
+                        *(short*)(bytePointer + 4) = (short)0;
+                        return;
+                    case 7:
+                        *(int*)bytePointer = (int)0;
+                        *(short*)(bytePointer + 4) = (short)0;
+                        *(bytePointer + 6) = (byte)0;
+                        return;
+                    case 8:
+                        *(long*)bytePointer = (long)0;
```

Comment:
Agree - if should use BIT64 ifdefs similar to what Buffer.Memmove implementation does.